src/tests/ftest/control/dmg_pool_query_ranks.py

```diff
@@ -80,58 +87,77 @@ def test_pool_query_ranks_mgmt(self):
-                data['response'].get('enabled_ranks'), enabled_ranks,
-                "Invalid enabled_ranks field: want={}, got={}".format(
-                    enabled_ranks, data['response'].get('enabled_ranks')))
-            self.assertListEqual(
-                data['response'].get('disabled_ranks'), disabled_ranks,
-                "Invalid disabled_ranks field: want={}, got={}".format(
-                    disabled_ranks, data['response'].get('disabled_ranks')))
-
-            self.log.debug("Waiting for pool to be rebuild")
-            self.pool.wait_for_rebuild_to_start()
-            self.pool.wait_for_rebuild_to_end()
+        self.log.debug("Checking enabled ranks state information")
+        data = self.dmg.pool_query(self.pool.identifier, show_enabled=True)
+        self.assertListEqual(
+            data['response'].get('enabled_ranks'), enabled_ranks,
+            "Invalid enabled_ranks field: want={}, got={}".format(
+                enabled_ranks, data['response'].get('enabled_ranks')))
+        self.assertListEqual(
+            data['response'].get('disabled_ranks'), disabled_ranks,
+            "Invalid disabled_ranks field: want={}, got={}".format(
+                disabled_ranks, data['response'].get('disabled_ranks')))
+
+        self.log.debug("Waiting for pool to be rebuild")
+        self.pool.wait_for_rebuild_to_start()
+
+        # kill second rank.
+        self.server_managers[0].stop_ranks([suspect_rank], self.d_log)
+        time.sleep(20)
+
+        self.log.debug("Checking suspect ranks state information")
+        data = self.dmg.pool_query(self.pool.identifier, health_only=True)
+        self.assertListEqual(
+            data['response'].get('suspect_ranks'), suspect_ranks,
+            "Invalid enabled_ranks field: want={}, got={}".format(
+                suspect_ranks, data['response'].get('suspect_ranks')))
+        self.assertListEqual(
+            data['response'].get('disabled_ranks'), disabled_ranks,
+            "Invalid disabled_ranks field: want={}, got={}".format(
+                disabled_ranks, data['response'].get('disabled_ranks')))
+
+        self.server_managers[0].start_ranks([suspect_rank], self.d_log)
+        time.sleep(10)
+        data = self.dmg.pool_query(self.pool.identifier, health_only=True)
+        self.assertListEqual(
+            data['response'].get('suspect_ranks'), [],
+            "Invalid enabled_ranks field: want={}, got={}".format(
+                [], data['response'].get('suspect_ranks')))
+
+        self.log.debug("Reintegrating rank %d", exclude_rank)
+        cmd_succeed = False
+        for _ in range(3):
+            try:
+                result = self.pool.reintegrate(exclude_rank)
+                cmd_succeed = True
+                break
+            except CommandFailure:
+                self.log.debug("dmg command failed retry")
+            time.sleep(3)
+
+        self.assertTrue(cmd_succeed, "pool reintegrate failed: {}".format(result))
```

Comment:
If `self.pool.reintegrate(exclude_rank)` failed above and raised an exception, `result` will not be set. I think in this case we don't need to log it, since the output will be in the test log
```suggestion
        self.assertTrue(cmd_succeed, "pool reintegrate failed")
```

Reply:
Fixed.